### ISSUE TITLE
Ripper error when using Ruby 2.5's `rescue` inside `do`/`end` blocks without `begin` syntax

### ISSUE THREAD

**Issue by koic** [jruby/jruby#6642]
**Environment Information**
  
Provide at least:
* JRuby version (`jruby -v`) and command line (flags, JRUBY_OPTS, etc)

```console
% jruby -v
jruby 9.2.17.0 (2.5.8) 2021-03-29 84d363da97 Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit [darwin-x86_64]
```

* Operating system and platform (e.g. `uname -a`)

```console
% uname -a
Darwin atelier.local 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64
```

**Expected Behavior**

The following is an example:

```ruby
# example.rb
require 'ripper'

class RipperParser < Ripper
  def on_parse_error(message)
    raise message
  end
end

RipperParser.new(<<~RUBY).parse
  foo do
    bar do
    end
  rescue NameError
    converted << nil
  end
RUBY
```

This is behavior of MRI 2.5.8.

```console
% ruby -v
ruby 2.5.8p224 (2020-03-31 revision 67882) [x86_64-darwin17]
% ruby example.rb
```

No errors.

**Actual Behavior**

This is behavior of JRuby 9.2.17.0.

```console
% ruby -v
jruby 9.2.17.0 (2.5.8) 2021-03-29 84d363da97 Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit [darwin-x86_64]
% ruby example.rb
RuntimeError: syntax error, unexpected keyword_rescue

  on_parse_error at example.rb:6
           parse at org/jruby/ext/ripper/RubyRipper.java:347
          <main> at example.rb:10
```

Unexpectedly, JRuby 9.2.17.0 raises the `RuntimeError`.

I encountered this issue in YARD on which RuboCop depends.
https://github.com/rubocop/rubocop/pull/9648

**Comment by koic** [comment]
It still reproduces in JRuby 9.3.0.0.

```console
% ruby -v
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-darwin19]
% ruby example.rb # no errors.

% ruby -v
jruby 9.3.0.0 (2.6.8) 2021-09-17 85c20e780f Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit 
% ruby example.rb
RuntimeError: syntax error, unexpected rescue

  on_parse_error at example.rb:6
           parse at org/jruby/ext/ripper/RubyRipper.java:342
          <main> at example.rb:10
```

**Comment by koic** [comment]
This issue has been resolved in JRuby 9.4:

```console
$ ruby -v
jruby 9.4.2.0 (3.1.0) 2023-03-08 90d2913fda Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit [x86_64-darwin]
$ ruby example.rb # No errors
```

Thank you!
